src/commonMain/kotlin/fr/acinq/lightning/wire/LightningMessages.kt

```diff
@@ -154,11 +155,22 @@ data class EncryptedChannelData(val data: ByteVector) {
     }
 }
 
-interface HasEncryptedChannelData : LightningMessage {
+interface HasEncryptedChannelData : HasChannelId {
     val channelData: EncryptedChannelData
-    fun withChannelData(data: ByteVector): HasEncryptedChannelData = withChannelData(EncryptedChannelData(data))
-    fun withChannelData(ecd: EncryptedChannelData): HasEncryptedChannelData = if (ecd.isEmpty()) this else withNonEmptyChannelData(ecd)
     fun withNonEmptyChannelData(ecd: EncryptedChannelData): HasEncryptedChannelData
+    fun withChannelData(data: ByteVector, logger: Logger? = null): HasEncryptedChannelData = withChannelData(EncryptedChannelData(data), logger)
+    fun withChannelData(ecd: EncryptedChannelData, logger: Logger? = null): HasEncryptedChannelData {
+        // The maximum size for lightning messages is 65535 bytes, so we only include a channel backup when we have enough space available.
+        // Our peer needs to be able to send the backup back in their channel_reestablish message, and we don't know the size of that message.
+        // We use an upper bound of 64000 bytes, hoping that it will be enough to make it work in all cases.
+        val sizeWithoutBackup = write().size
+        return if (!ecd.isEmpty() && ecd.data.size() + sizeWithoutBackup <= 64_000) {
+            withNonEmptyChannelData(ecd)
+        } else {
+            logger?.warning { "c:${this.channelId} could not include encrypted backup in ${this::class.simpleName}: too large (${ecd.data.size()} bytes)" }
```

Comment:
On an actual restore with an eclair peer, this prints `too large (0 bytes)`:

```
18:11:27.648 INFO  f.a.l.c.ChannelContext - c:c84d2256d3441abb0e0782116e1ff5b262175bc85a6c90daad7c93063987d248 channel_reestablish includes a peer backup
18:11:27.649 INFO  f.a.l.c.ChannelContext - c:c84d2256d3441abb0e0782116e1ff5b262175bc85a6c90daad7c93063987d248 ignoring their older backup
18:11:27.661 WARN  f.a.l.c.ChannelContext - c:c84d2256d3441abb0e0782116e1ff5b262175bc85a6c90daad7c93063987d248 could not include encrypted backup in ChannelReestablish: too large (0 bytes)
```

There is another faulty log `ignoring their older backup` that I will fix in #401.

Reply:
Good catch, I'm printing that error also for the case where the backup is empty, it's a mistake.

Reply:
> There is another faulty log ignoring their older backup that I will fix in https://github.com/ACINQ/lightning-kmp/pull/401.

Done in [bbc3456](https://github.com/ACINQ/lightning-kmp/pull/401/commits/bbc3456f00cffe2e580a175e5d65e62267aa6132).

Reply:
I avoid logging anything on empty backups in https://github.com/ACINQ/lightning-kmp/pull/410/commits/973e9ce48d2708a8f8d5f07b31ab71d5d8cb7ef7